test/gtest/deepbench_lstm.cpp

```diff
@@ -0,0 +1,193 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2023 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#include <utility>
+
+#include "lstm.hpp"
+#include "get_handle.hpp"
+#include <miopen/env.hpp>
+#include <gtest/gtest_common.hpp>
+#include <gtest/gtest.h>
+
+MIOPEN_DECLARE_ENV_VAR_BOOL(MIOPEN_TEST_DEEPBENCH)
+MIOPEN_DECLARE_ENV_VAR_BOOL(MIOPEN_TEST_FLOAT)
+MIOPEN_DECLARE_ENV_VAR_BOOL(MIOPEN_TEST_HALF)
+
+namespace deepbench_lstm {
+using EnvType = std::tuple<std::pair<miopen::env::MIOPEN_TEST_DEEPBENCH, std::string_view>,
+                           std::pair<miopen::env::MIOPEN_TEST_FLOAT, std::string_view>,
+                           std::pair<miopen::env::MIOPEN_TEST_HALF, std::string_view>>;
+
+static bool Skip(miopenDataType_t prec)
+{
+    bool flag = !miopen::IsEnabled(ENV(MIOPEN_TEST_DEEPBENCH));
```

Comment:
This function should allow the test to run
- if `MIOPEN_TEST_FLOAT_ARG` is equal "--float" or empty
- and if `MIOPEN_TEST_DEEPBENCH` is enabled.

Just like the original test.

Reply:
[resolved]